### PR TITLE
Add Similar Petitions Suggestion while creating a petition

### PR DESCRIPTION
**Pull request by errhythm**
Added a Similar Petitions card, which will prompt the user if their petition already exists. Used the Stopword package to remove the stopwords from the title, which would increase the accuracy of the search. 

image

https://github.com/user-attachments/assets/1a054b1d-ebea-46a3-ae96-fefbe157eb4f

Please review and suggest any necessary changes; I am still a newbie. Also, I found that the suggestion result could be increased with Postgres [pg_trgm](https://www.postgresql.org/docs/current/pgtrgm.html), but I need to familiarize myself with it.

### CODE REVIEW

**Comment by errhythm** [comment]
Updated the search accuracy and changed the card UI. 

image

**Review by getanwar — CHANGES_REQUESTED** [review]
Please check my comments :)

File: apps/jonogon-core/src/api/trpc/procedures/petitions/listing/list-petitions.mts [hunk 1]
```diff
@@ -394,3 +396,72 @@ export const listSuggestedPetitions = protectedProcedure
...
+                .where('vote', '=', -1)
+                .executeTakeFirst();
+
+            return {
+                ...petition,
+                upvotes: Number(upvotes?.count || 0),
+                downvotes: Number(downvotes?.count || 0),
+                match_count: petition.title
+                    ? keywords.filter(keyword =>
+                        petition.title!.toLowerCase().includes(keyword.toLowerCase())
```
[the remark this answers is not in the document]
Comment (errhythm) [reply apps/jonogon-core/src/api/trpc/procedures/petitions/listing/list-petitions.mts:451]: My bad. You are right. I meant to use (`?.`) instead.

File: apps/jonogon-web-next/src/components/custom/SimilarPetitionsSuggestions.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import { useState, useEffect, useCallback } from 'react';
+import { trpc } from '@/trpc/client';
+import Link from 'next/link';
+import { Card, CardHeader, CardTitle, CardContent } from '@/components/ui/card';
+import { Button } from '@/components/ui/button';
+import { AlertCircle, ArrowRight, ThumbsUp, ThumbsDown } from 'lucide-react';
+import { debounce } from 'lodash';
```
[the remark this answers is not in the document]
Comment (errhythm) [reply apps/jonogon-web-next/src/components/custom/SimilarPetitionsSuggestions.tsx:7]: Got it. I will fix it as well. Thank you.

**Comment by errhythm** [comment]
> Please check my comments :)

Thanks. I have fixed it according to your suggestions.

**Review by getanwar — APPROVED** [review]

**Pull request merged**